eZ/Publish/Core/Repository/ProxyFactory/ProxyDomainMapper.php

```diff
@@ -0,0 +1,198 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\Core\Repository\ProxyFactory;
+
+use eZ\Publish\API\Repository\Repository;
+use eZ\Publish\API\Repository\Values\Content\Content;
+use eZ\Publish\API\Repository\Values\Content\ContentInfo;
+use eZ\Publish\API\Repository\Values\Content\Language;
+use eZ\Publish\API\Repository\Values\Content\Location;
+use eZ\Publish\API\Repository\Values\Content\Section;
+use eZ\Publish\API\Repository\Values\ContentType\ContentType;
+use eZ\Publish\API\Repository\Values\ContentType\ContentTypeGroup;
+use eZ\Publish\API\Repository\Values\User\User;
+use ProxyManager\Factory\LazyLoadingValueHolderFactory;
+use ProxyManager\Proxy\LazyLoadingInterface;
+
+/**
+ * @internal
+ */
+final class ProxyDomainMapper implements ProxyDomainMapperInterface
+{
+    /** @var \eZ\Publish\API\Repository\Repository */
+    private $repository;
+
+    /** @var \ProxyManager\Factory\LazyLoadingValueHolderFactory */
+    private $factory;
+
+    public function __construct(Repository $repository, LazyLoadingValueHolderFactory $lazyLoadingValueHolderFactory)
+    {
+        $this->repository = $repository;
+        $this->factory = $lazyLoadingValueHolderFactory;
+    }
+
+    public function createContentProxy(
+        int $contentId,
+        array $prioritizedLanguages = Language::ALL,
+        bool $useAlwaysAvailable = true
+    ): Content {
+        $initializer = function (
+            &$wrappedObject, LazyLoadingInterface $proxy, $method, array $parameters, &$initializer
+        ) use ($contentId, $prioritizedLanguages, $useAlwaysAvailable): bool {
+            $initializer = null;
+            $wrappedObject = $this->repository->getContentService()->loadContent(
+                $contentId,
+                $prioritizedLanguages,
+                null,
+                $useAlwaysAvailable
+            );
+
+            return true;
+        };
+
+        return $this->factory->createProxy(Content::class, $initializer);
```

Comment:
It generates an issue
> Return value is expected to be '\eZ\Publish\API\Repository\Values\Content\Content', `'\ProxyManager\Proxy\VirtualProxyInterface'` returned

I think for now those methods need to have `@return` PhpDoc.